pkgs/by-name/wi/wivrn/package.nix

```diff
@@ -85,62 +117,121 @@ stdenv.mkDerivation (finalAttrs: {
     fi
   '';
 
-  nativeBuildInputs = [
-    cmake
-    git
-    glslang
-    pkg-config
-    python3
-  ] ++ lib.optionals cudaSupport [ autoAddDriverRunpath ];
+  nativeBuildInputs =
+    [
+      cmake
+      doxygen
+      git
+      glib
+      glslang
+      librsvg
+      pkg-config
+      python3
+      qt6.wrapQtAppsHook
+    ]
+    ++ lib.optionals cudaSupport [
+      autoAddDriverRunpath
+    ];
 
-  buildInputs = [
-    avahi
-    boost
-    cli11
-    eigen
-    ffmpeg
-    freetype
-    glm
-    harfbuzz
-    libdrm
-    libGL
-    libva
-    libX11
-    libXrandr
-    libpulseaudio
-    nlohmann_json
-    onnxruntime
-    openxr-loader
-    pipewire
-    shaderc
-    spdlog
-    systemd
-    udev
-    vulkan-headers
-    vulkan-loader
-    vulkan-tools
-    x264
-  ] ++ lib.optionals cudaSupport [ cudaPackages.cudatoolkit ];
+  buildInputs =
+    [
+      avahi
+      boost
+      bluez
+      cjson
+      cli11
+      dbus
+      eigen
+      elfutils
+      ffmpeg
+      freetype
+      glib
+      glm
+      gst_all_1.gst-plugins-base
+      gst_all_1.gstreamer
+      harfbuzz
+      hidapi
+      libbsd
+      libdrm
+      libdwg
+      libGL
+      libjpeg
+      libmd
+      libnotify
+      librealsense
+      libsurvive
+      libunwind
+      libusb1
+      libuvc
+      libva
+      libX11
+      libXrandr
+      libpulseaudio
+      nlohmann_json
+      opencv4
+      openhmd
+      openvr
+      openxr-loader
+      onnxruntime
+      orc
+      pipewire
+      qt6.qtbase
+      qt6.qttools
+      SDL2
+      shaderc
+      spdlog
+      systemd
+      udev
+      vulkan-headers
+      vulkan-loader
+      wayland
+      wayland-protocols
+      wayland-scanner
+      x264
+    ]
+    ++ lib.optionals cudaSupport [
+      cudaPackages.cudatoolkit
+    ];
 
   cmakeFlags = [
+    (lib.cmakeBool "WIVRN_USE_NVENC" cudaSupport)
     (lib.cmakeBool "WIVRN_USE_VAAPI" true)
+    (lib.cmakeBool "WIVRN_USE_VULKAN" true)
     (lib.cmakeBool "WIVRN_USE_X264" true)
-    (lib.cmakeBool "WIVRN_USE_NVENC" cudaSupport)
-    (lib.cmakeBool "WIVRN_USE_SYSTEMD" true)
     (lib.cmakeBool "WIVRN_USE_PIPEWIRE" true)
     (lib.cmakeBool "WIVRN_USE_PULSEAUDIO" true)
+    (lib.cmakeBool "WIVRN_FEATURE_STEAMVR_LIGHTHOUSE" true)
     (lib.cmakeBool "WIVRN_BUILD_CLIENT" false)
+    (lib.cmakeBool "WIVRN_BUILD_DASHBOARD" true)
+    (lib.cmakeBool "WIVRN_CHECK_CAPSYSNICE" false)
     (lib.cmakeBool "WIVRN_OPENXR_INSTALL_ABSOLUTE_RUNTIME_PATH" true)
     (lib.cmakeBool "FETCHCONTENT_FULLY_DISCONNECTED" true)
+    (lib.cmakeFeature "GIT_DESC" "${finalAttrs.version}")
     (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_MONADO" "${finalAttrs.monado}")
```

Comment:
Should we include `"-DCUDA_TOOLKIT_ROOT_DIR=${prev.cudaPackages.cudatoolkit}"` since that's what me and @Mistyttm found to make NVENC work with WiVRn?

Reply:
I'm not sure, that issue only appeared with the way the package was imported and I don't think it's necessarily related to WiVRn. @Scrumplex

Reply:
I used the overlay that Mistyttm's overlay too and it still requires that flag to be set (CMake itself ends the build saying `CMake Error  Specify CUDA_TOOLKIT_ROOT_DIR`".. I've had other packages do similarly explicitly because I enabled the `cudaSupport` flag via `nixpkgs.config.cudaSupport = true`

Reply:
Did some more testing to confirm my suspistion. Yeah we do in fact need that cuda flag, I try to run WiVRN without the flag and without the override and it starts complaining about not having the monado nvidia layers installed (despite I have it added into the extraPackages) aswell as NVENC not being enabled for the build